lib/usbmux/transformer/usbmux-decoder.js

```diff
@@ -0,0 +1,32 @@
+import { parse } from '../../util/plist';
+import Stream from 'stream';
+
+const HEADER_LENGTH = 16;
+
+class UsbmuxDecoder extends Stream.Transform {
+
+  constructor () {
+    super({ objectMode: true });
+  }
+
+  _transform (data, encoding, callback) {
+    this._decode(data);
+    callback();
+  }
+
+  _decode (data) {
+    let header = {
+      length: data.readUInt32LE(0),
+      version: data.readUInt32LE(4),
+      type: data.readUInt32LE(8),
+      tag: data.readUInt32LE(12)
+    };
+
+    let payload = data.slice(HEADER_LENGTH, data.length);
+    this.push({ header, payload: parse(payload) });
```

Comment:
should we check if the payload is parseable?

Reply:
I'm cool with throwing an exception when it is not parseable. In theory, it should always be parseable for usbmux messages